packages/luna/tsconfig.json

```diff
@@ -1,27 +1,39 @@
 {
   "compilerOptions": {
-    "target": "ES2020",
-    "useDefineForClassFields": true,
-    "lib": ["ES2020", "DOM", "DOM.Iterable"],
+    /* Basic Options */
+    "target": "ES6",
     "module": "ESNext",
-    "skipLibCheck": true,
+    "lib": ["ES2020", "DOM", "DOM.Iterable"],
+    "jsx": "react-jsx",
+
+    /* Output Options */
+    "rootDir": "src",
+    "outDir": "dist",
+    "declarationMap": true,
+    "declaration": true,
 
-    /* Bundler mode */
+    /* Module Resolution */
     "moduleResolution": "bundler",
-    "allowImportingTsExtensions": true,
-    "resolveJsonModule": true,
-    "isolatedModules": true,
-    "noEmit": true,
-    "jsx": "react-jsx",
+    "baseUrl": ".",
     "paths": {
       "@/*": ["./src/*"]
     },
-    /* Linting */
+    "resolvePackageJsonExports": true,
+    "resolvePackageJsonImports": true,
+    "resolveJsonModule": true,
+
+    /* Type Checking and Linting */
+    "noImplicitAny": true,
+    "allowUmdGlobalAccess": true,
     "strict": true,
     "noUnusedLocals": true,
     "noUnusedParameters": true,
-    "noFallthroughCasesInSwitch": true
+    "noFallthroughCasesInSwitch": true,
+    "useDefineForClassFields": true,
+    "skipLibCheck": true,
+    "allowSyntheticDefaultImports": true,
+    "forceConsistentCasingInFileNames": true,
+    "isolatedModules": true
   },
-  "include": ["src"],
-  "references": [{ "path": "./tsconfig.node.json" }]
+  "include": ["src", ".storybook", "src/**/*.stories.tsx"]
```

Comment:
This way we will bundle storybooks along side our components which something we don't want to do. 

To resolve this issue you should do the following. 

In `tsconfig.json`  keep `include` as it is. 

```json
  "include": ["src", ".storybook", "src/**/*.stories.tsx"]
```
Create another config file and call it `tsconfig.build.json`. It should extend `tsconfig.json`. It should include the `src/` only. This way we will only include the components in our build. 

```ts
{
  "extends": "./tsconfig.json",
  "include": ["src/"], 
  "exclude": ["src/**/*.stories.txs"],
}
```

When building the components make sure to use `tsconfig.build.json` and not the default `tsconfig.json`.